### PR TITLE
Use OpenGL projection matricies over Direct3D for better depth precision

### PR DESCRIPTION
**Pull request by haxiomic**
**Needs discussion before merge**

**TL;DR**: Small adjustment to projection matrices = a little less z-fighting

[WebGL's normalized device coordinates span the unit cube](http://www.songho.ca/opengl/gl_projectionmatrix.html), Direct3D normalized device coordinates are the same execpt z spans from 0 to 1 instead of -1 to 1.

Babylon uses projection matrices suited for Direct3D, consequently if a vertex is at the near plane a value of 0.5 will be written to the depth buffer and if a vertex is at the far plane 1.0 will be written to the depth buffer, so we're missing out on the full bit-depth available to us.

The fix (in this PR) is to adjust the projection matrices so when z = near plane, -1.0 is written and when z = far plane, 1.0 is written.

I noticed `PerspectiveFovWebVRToRef` has this adjustment but it's commented out.

In most cases this change will have no impact (other than increasing depth precision), but some code may rely on the old values (shadow techniques like VSM may look different for example) so we need input from people who know more about where this change may impact

### CODE REVIEW

**Comment by deltakosh** [comment]
Excellent! It was on my todo list since ages
I'll proceed to merge and do fixes as required

**Comment by deltakosh** [comment]
Please credits yourself in the what's new for 2.6 :)

**Comment by wyklion** [comment]
i see something wrong here.... 
code changing made my hero model display alpha wrong...

**Comment by deltakosh** [comment]
alpha? This is not related to shaders or colors. Can you provide a repro on PG?

**Comment by wyklion** [comment]
no... too hard to make it. i try in my project. my rotateCamera's radius = 1. the model is very closed to screen. and some skin looks like disppear or look otherside. and when i set radius = 3. whole model skin can be seen. so, it's a z calculate problem.

**Comment by wyklion** [comment]
maybe it's right. just screen cut model.

**Comment by haxiomic** [comment]
It'll be the near plane frustum culling - with the old matricies if a vertex is as at the near plane of the camera frustum, it's transformed into clip space as half way between the near at the far planes, so it can still get closer to the camera before being culled.

Try reducing your current near plane by a factor of 2
